spec/lib/carto/legend_migrator_spec.rb

```
@@ -0,0 +1,408 @@
+              "visible" => true,
+              "value" => "#FF7F00"
+            },
+            {
+              "name" => 900,
+              "visible" => true,
+              "value" => "#CAB2D6"
+            },
+            {
+              "name" => 90000,
+              "visible" => true,
+              "value" => "#6A3D9A"
+            },
+            {
+              "name" => "Others",
+              "visible" => true,
+              "value" => "#DDDDDD"
+            }
+          ]
+        }
+      end
+
+      let(:old_custom) do
+        {
+          "type" => "custom",
+          "show_title" => true,
+          "title" => "",
+          "template" => "",
+          "visible" => true,
+          "items" => [
+            {
+              "name" => "preta",
+              "visible" => true,
+              "value" => "#41006D",
+              "sync" => true
+            },
+            {
+              "name" => "Untitled",
+              "visible" => true,
+              "value" => "#3E7BB6",
+              "sync" => true
+            },
+            {
+              "name" => "patata",
+              "visible" => true,
+              "value" => "#cccccc",
+              "sync" => true
+            },
+            {
+              "name" => "Untitled",
+              "visible" => true,
+              "value" => "#cccccc",
+              "sync" => true
+            }
+          ]
+        }
+      end
+
+      it 'omits badly formatted colors' do
+        truncated = old_category.dup
+        truncated['items'].first['value'] = '#fatal#fatal#fatal'
+
+        @old_legend = truncated
+      end
+
+      it 'migrates old category to new custom' do
+        @old_legend = old_category
+      end
+
+      it 'migrates old custom to new custom' do
+        @old_legend = old_custom
+      end
+
+      after(:each) do
+        new_legend = Carto::LegendMigrator.new(@layer.id, @old_legend).build
+
+        new_legend.type.should eq 'custom'
```

Have you tried if asserts work in `after` code? I know exceptions are silently eaten.